File: 3629-total-characters-in-string-after-transformations-i/3629-total-characters-in-string-after-transformations-i.py
class Solution(object):
    def lengthAfterTransformations(self, s, t):
        MOD=int(1e9+7)
        freq=[0]*26
        for ch in s: freq[ord(ch)-ord('a')]+=1
        for _ in range(t):
           x=freq[25]
           for i in range(25,0,-1):
              freq[i]=freq[i-1]
           freq[0]=x
           freq[1]=(freq[1]+x)%MOD
        
        res=sum(freq)%MOD
        return res
        

File: 3629-total-characters-in-string-after-transformations-i/3629-total-characters-in-string-after-transformations-i.cpp
class Solution {
public:
    int MOD=1e9+7;
    int lengthAfterTransformations(string s, int t) {
        vector<long long>freq(26,0);
        for(int i=0;i<s.size();i++) freq[s[i]-'a']++;
        while(t--){
           int x=freq[25];
           for(int i=25;i>0;i--) freq[i]=freq[i-1];
           freq[0]=x;
           freq[1]=(freq[1]+x)%MOD;
        }
        
        int res=0;
        for(int i=0;i<26;i++) res=(res+freq[i])%MOD;
        return res;
    }
};